frontend/src/plugins/impl/plotly/PlotlyPlugin.tsx

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2023 Marimo. All rights reserved. */
+import { z } from "zod";
+
+import { IPluginProps } from "@/plugins/types";
+
+import Plot, { Figure } from "react-plotly.js";
+import { IStatelessPlugin } from "@/plugins/stateless-plugin";
+import { T } from "vitest/dist/reporters-5f784f42";
+import { Logger } from "@/utils/Logger";
+
+import "./plotly.css";
+
+interface Data {
+  figure: Figure;
+}
+
+export class PlotlyPlugin implements IStatelessPlugin<Data> {
+  tagName = "marimo-plotly";
+
+  validator = z.object({
+    figure: z
+      .object({})
+      .passthrough()
+      .transform((spec) => spec as unknown as Figure),
+  });
+
+  render(props: IPluginProps<T, Data>): JSX.Element {
+    return <PlotlyComponent {...props.data} host={props.host} />;
+  }
+}
+
+interface PlotlyPluginProps extends Data {
+  host: HTMLElement;
+}
+
+export const PlotlyComponent = ({ figure, host }: PlotlyPluginProps) => {
+  return (
+    <Plot
+      {...figure}
+      layout={{
+        ...figure.layout,
+        autosize: true,
```

Comment:
Can the user still manually specify a width and height in Python, or will this override it? Sorry, I'm not familiar with the Plotly.js API. Sometimes manually specified width/height are important (eg, if the data you're plotting requires a certain aspect ratio)

Reply:
the docs says that this is only applied if the height, width is `undefined`. ill double check